hexa/pipeline_templates/models.py

```diff
@@ -77,6 +78,20 @@ def upgrade_pipeline(
             principal, pipeline.workspace, pipeline
         )
 
+    def new_versions(self, pipeline: Pipeline) -> list["PipelineTemplateVersion"]:
```

Comment:
I have the impression that adding a verb in front of `new_versions` would clarify how to use this

```suggestion
    def get_new_versions(self, pipeline: Pipeline) -> list["PipelineTemplateVersion"]:
```
?